wordpress-mobile/WordPress-Android
# Upgrade gradle to 7.1.1 & Android Gradle Plugin to 4.2.2

This PR upgrades Gradle to 7.1.1 and Android Gradle Plugin to 4.2.2. The changes in the PR are minimal as all the necessary changes have been made in other PRs and already got merged. Besides the upgrades, the PR references the Gradle 7 branch of FluxC, gutenberg-mobile & stories-android libraries so that it can all be tested in this PR.

Note that I am leaving this PR as a draft because all Gradle 7 upgrade PRs need to be merged together. We'll merge all 3 libraries first and update the references before merging this PR.

**To test:**
* Smoke test the app - login & editor should be enough
* Use composite build for fluxc & gutenberg-mobile libraries (following the instructions in [local-builds.gradle-example](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/local-builds.gradle-example) file)  and smoke test the app again. Make sure you're on the Gradle 7 branches for [fluxc](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2084) & [gutenberg-mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3833). Don't forget to run `npm install & npm run start:reset` from gutenberg-mobile folder to start the metro server. You'll also need to do `nvm use` from `gutenberg-mobile/gutenberg` folder before running `npm install` or you might have errors.

## Regression Notes
1. Potential unintended areas of impact
Probably N/A - The build file changes should only impact whether the project builds or not. However, since Android Gradle Plugin is updated, it's hard to be sure that nothing else will be impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests and current CI checks mostly around the build

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
